internal/dockercompose/client.go

```diff
@@ -33,8 +33,20 @@ func NewDockerComposeClient() DockerComposeClient {
 	return &cmdDCClient{}
 }
 
-func (c *cmdDCClient) Up(ctx context.Context, configPath string, serviceName model.TargetName, stdout, stderr io.Writer) error {
-	cmd := exec.CommandContext(ctx, "docker-compose", "-f", configPath, "up", "--no-deps", "--build", "-d", serviceName.String())
+func (c *cmdDCClient) Up(ctx context.Context, configPath string, serviceName model.TargetName, shouldBuild bool, stdout, stderr io.Writer) error {
+	args := []string{"-f", configPath, "up", "--no-deps", "-d"}
+	if shouldBuild {
+		args = append(args, "--build")
+	} else {
+		// !shouldBuild implies that Tilt will take care of building, which implies that
+		// we should recreate container so that we pull the new image
+		// NOTE(maia): this is maybe the WRONG thing to do if we're deploying a service
```

Comment:
for what it's worth, this is another reason why our k8s-based builder uses content-based tags: it's a way to make sure kubernetes creates a new pod iff the contents changed.